src/main/java/servlets/module/challenge/XxeChallenge1.java

```diff
@@ -0,0 +1,171 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.<br/>
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with the Security Shepherd project.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author ismisepaul
+ */
+public class XxeChallenge1
+        extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+    private static Logger log = Logger.getLogger(XxeChallenge1.class);
+    private static final String LEVEL_NAME = "XXE Challenge 1";
+    private static final String LEVEL_HASH = "ac8f3f6224b1ea3fb8a0f017aadd0d84013ea2c80e232c980e54dd753700123e";
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+
+        String ApplicationRoot = getServletContext().getRealPath("");
+        String moduleId = Getter.getModuleIdFromHash(ApplicationRoot, LEVEL_HASH);
+
+        //Setting IpAddress To Log and taking header for original IP if forwarded from proxy
+        ShepherdLogManager.setRequestIp(request.getRemoteAddr(), request.getHeader("X-Forwarded-For"));
+        log.debug(LEVEL_NAME + " Servlet Accessed");
+        PrintWriter out = response.getWriter();
+        out.print(getServletInfo());
+
+        //Translation Stuff
+        Locale locale = new Locale(Validate.validateLanguage(request.getSession()));
+        ResourceBundle errors = ResourceBundle.getBundle("i18n.servlets.errors", locale);
+        ResourceBundle bundle = ResourceBundle.getBundle("i18n.servlets.challenges.injection.xxe1", locale);
+
+        try {
+            HttpSession ses = request.getSession(true);
+            if (Validate.validateSession(ses))
+            {
+                if (Getter.isModuleOpen(getServletContext().getRealPath(""), moduleId))
+                {
+                    ShepherdLogManager.setRequestIp(request.getRemoteAddr(), request.getHeader("X-Forwarded-For"),
+                            ses.getAttribute("userName").toString());
+                    log.debug(LEVEL_NAME + " accessed by: " + ses.getAttribute("userName").toString());
+                    Cookie tokenCookie = Validate.getToken(request.getCookies());
+                    Object tokenHeader = request.getHeader("csrfToken").toString();
+
+                    if (Validate.validateTokens(tokenCookie, tokenHeader)) {
+                        InputStream json = request.getInputStream();
+                        String emailAddr = readJson(json);
+                        log.debug("Email Addr: " + emailAddr);
+
+                        String htmlOutput = new String();
+
+                        if (emailAddr == null) {
+                            htmlOutput += "<p>" + bundle.getString("response.blank.email") + "</p>";
+                            out.write(htmlOutput + emailAddr);
```

Comment:
Should emailAddr be encoded for HTML output?

Reply:
```suggestion
                            out.write(htmlOutput + emailAddr);
```
is done at line 89: emailAddr = Encode.forHtml(emailAddr);